src/avro_fingerprint.erl

```diff
@@ -0,0 +1,335 @@
+%% coding: latin-1
+%%%-----------------------------------------------------------------------------
+%%%
+%%% Copyright (c) 2018 Klarna AB
+%%%
+%%% This file is provided to you under the Apache License,
+%%% Version 2.0 (the "License"); you may not use this file
+%%% except in compliance with the License.  You may obtain
+%%% a copy of the License at
+%%%
+%%%   http://www.apache.org/licenses/LICENSE-2.0
+%%%
+%%% Unless required by applicable law or agreed to in writing,
+%%% software distributed under the License is distributed on an
+%%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%%% KIND, either express or implied.  See the License for the
+%%% specific language governing permissions and limitations
+%%% under the License.
+%%%
+%%% @author Jake Morrison <jake@cogini.com>
+%%%
+%%% @doc
+%%% Implement the Avro CRC 64 fingerprint algorithm.
+%%%
+%%% The algorithm is defined with this Java code:
+%%%
+%%%  ```
+%%%   static long fingerprint64(byte[] buf) {
+%%%       if (FP_TABLE == null) initFPTable();
+%%%       long fp = EMPTY;
+%%%       for (int i = 0; i < buf.length; i++)
+%%%           fp = (fp >>> 8) ^ FP_TABLE[(int)(fp ^ buf[i]) & 0xff];
+%%%       return fp;
+%%%   }
+%%%
+%%%   static long EMPTY = 0xc15d213aa4d7a795L;
+%%%   static long[] FP_TABLE = null;
+%%%
+%%%   static void initFPTable() {
+%%%       FP_TABLE = new long[256];
+%%%       for (int i = 0; i < 256; i++) {
+%%%           long fp = i;
+%%%           for (int j = 0; j < 8; j++)
+%%%               fp = (fp >>> 1) ^ (EMPTY & -(fp & 1L));
+%%%           FP_TABLE[i] = fp;
+%%%           // Generate the lookup table used in this code
+%%%           System.out.format("fp_table(%d) -> 16#%s;%n", i, Long.toHexString(fp));
+%%%       }
+%%%   }
+%%%  '''
+%%% @end
+%%%
+%%% @reference See <a href="https://avro.apache.org/docs/1.8.2/spec.html#schema_fingerprints">The
+%%% Avro Spec</a> for more information.
+%%% @reference See <a href="https://avro.apache.org/docs/1.8.2/api/java/org/apache/avro/SchemaNormalization.html">The source</a>
+%%%-----------------------------------------------------------------------------
+-module(avro_fingerprint).
+
+%% API
+-export([crc64/1]).
```

Comment:
please also export a type `avro_fingerprint:crc64()` then ref-export it from `avro.erl` : `avro:fingerprint()`

Reply:
Fixed. I exported the type avro_fingerprint:crc64() (non_neg_integer()) and made a function in avro:crc64_fingerprint/1 which calls avro_fingerprint:crc64/1

Reply:
I mean export the type from avro.erl too.